integration_tests/e2e/order/contact-information/primary-address.submission.cy.ts

```diff
@@ -0,0 +1,131 @@
+import { v4 as uuidv4 } from 'uuid'
+import Page from '../../../pages/page'
+import PrimaryAddressPage from '../../../pages/order/contact-information/primary-address'
+import NotifyingOrganisationPage from '../../../pages/order/contact-information/notifyingOrganisation'
+import OrderSummaryPage from '../../../pages/order/summary'
+import SecondaryAddressPage from '../../../pages/order/contact-information/secondary-address'
+
+const mockOrderId = uuidv4()
+const apiPath = '/address'
+
+context('Contact information', () => {
+  context('Primary address', () => {
+    context('Submitting a valid response', () => {
+      beforeEach(() => {
+        cy.task('reset')
+        cy.task('stubSignIn', { name: 'john smith', roles: ['ROLE_EM_CEMO__CREATE_ORDER'] })
+
+        cy.task('stubCemoGetOrder', { httpStatus: 200, id: mockOrderId, status: 'IN_PROGRESS' })
+        cy.task('stubCemoSubmitOrder', {
+          httpStatus: 200,
+          id: mockOrderId,
+          subPath: apiPath,
+          response: {
+            addressType: 'PRIMARY',
+            addressLine1: '',
+            addressLine2: '',
+            addressLine3: '',
+            addressLine4: '',
+            postcode: '',
+          },
+        })
+
+        cy.signIn()
+      })
+
+      it('should submit a correctly formatted address submission', () => {
+        const page = Page.visit(PrimaryAddressPage, {
+          orderId: mockOrderId,
+          'addressType(primary|secondary|tertiary)': 'primary',
+        })
+
+        const validFormData = {
+          addressLine1: 'line 1',
+          addressLine2: 'line 2',
+          addressLine3: 'line 3',
+          addressLine4: 'line 3',
```

Comment:
Not super important, but presumably this string should be 'line 4'.
The same applies to each set of form data below this one.